src/components/ProjectPage/ProjectPage.test.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { ProjectPage } from './ProjectPage';
+import FillButton from '../Button/FillButton';
+
+import {
+  openProjectInEditor,
+  openProjectInFolder,
+} from '../../services/shell.service';
+
+jest.mock('../../services/shell.service');
+
+const initialState = {
+  projects: {
+    byId: {
+      'a-project': {
+        path: 'path/to/project',
+        guppy: {
+          id: 'a-project',
+          name: 'example',
+          type: 'create-react-app',
+          color: '#fff',
+        },
+      },
+    },
+    selectedId: 'a-project',
+  },
+};
+
+const project = {
+  id: 'a-project', // Todo: Check how the id is added in app to project prop
+  ...initialState.projects.byId['a-project'],
+};
+
+window.scroll = jest.fn();
+
+describe('ProjectPage component', () => {
+  let wrapper;
+  let instance;
+  let mockActions;
+
+  const loadingStates = ['loading', 'done', 'fail'];
+  const mountWithStatus = (status, actions) =>
+    shallow(
+      <ProjectPage
+        project={project}
+        dependenciesLoadingStatus={status}
+        {...actions}
+      />
+    );
+
+  beforeEach(() => {
+    mockActions = {
+      loadDependencyInfoFromDisk: jest.fn(),
+      reinstallDependencies: jest.fn(),
+    };
+    wrapper = mountWithStatus('loading', mockActions);
+    instance = wrapper.instance();
+  });
+
+  describe('Rendering cases', () => {
+    loadingStates.forEach(dependencyLoadingStatus => {
+      it(`should render project page (${dependencyLoadingStatus})`, () => {
```

Comment:
Value not matching for these cases, due to the isMac check in `getCopyForOpeningFolder` (platform.service.js). I'm on Mac so I'm getting "Open in Finder", not folder. Does it need to be mocked?

<img width="350" alt="screen shot 2019-02-06 at 10 01 05 pm" src="https://user-images.githubusercontent.com/17421347/52393415-c90ce700-2a5a-11e9-9e8c-6d46f548acd5.png">

Reply:
Oh, OK, I missed that one. I'll add a mock for it.

Reply:
OK, added the mock. Should work for you now. Please re-run your test.

To test that the serivce function is working we need to add a test to `platform.service.test` but I think we'll add that in a separate PR.

Reply:
Works great :)